frontend/components/deck/Deck.tsx

```diff
@@ -28,14 +29,7 @@ interface DeckProps {
 }
 
 export const FetchDeck = ({ deck, ...props }: DeckProps) => {
-	const fetcher = (url: RequestInfo | URL) =>
-		authedFetch(url, {
-			method: "GET",
-		}).then((res) => res?.json());
-	const { data: dueCards } = useSWR(
-		deck ? `/api/decks/${deck?.deckID}/dueCards` : null,
-		fetcher
-	);
+	const { dueCards } = useDueCards(deck?.deckID ?? "");
```

Comment:
Since deck should never be `undefined` for the fetch deck, consider creating new props for this component where deck is not optional

```suggestion
	const { dueCards } = useDueCards(deck.deckID);
```